### PR TITLE
Arm AdvSimd: Rename StoreVectorMxN to Store

### LINKED ISSUE #102340: Arm64: Rename AdvSimd/Sve Load/Store vector API names
### AdvSimd
1. Rename `LoadVector128xN` to `LoadNxVector128 `
   -  Consistent naming. See https://github.com/dotnet/runtime/issues/97831#issuecomment-1934714477
2. Rename `StoreVectorNxMAndZip` to `StoreVectorAndZip`
   - The type on which the API is operation on can be known from parameters. As such we do not need special naming for them.
3. Rename `StoreVectorNx2` to `Store`
   - Same reason as point 2) above.

### Sve
1. Rename `LoadNxVector` to `LoadNxAndUnzip`
   - The operation does the unzip, similar to how it is done in AdvSimd. Compare left side operation vs. "4th quadrant" of right side.
image

2. Rename `Store` to `StoreAndZip`
   - Same reason as point 1) above

Discussion: https://github.com/dotnet/runtime/pull/102180#discussion_r1602562814

### PR DESCRIPTION
**Pull request by SwapnilGaikwad**
Partially (item 3) fixes: https://github.com/dotnet/runtime/issues/102340

### CODE REVIEW

**Comment by SwapnilGaikwad** [comment]
Hi @kunalspathak , this addresses the final (item 3) from https://github.com/dotnet/runtime/issues/102340.
Didn't change mono bits. 

@a74nh

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs [hunk 1]
```diff
@@ -1885,7 +1884,6 @@
...
     ("VecBinOpTest.template",             new Dictionary<string, string> { ["TestName"] = "SubtractSaturate_Vector64_Int32",                                                                       ["Isa"] = "AdvSimd",       ["LoadIsa"] = "AdvSimd", ["Method"] = "SubtractSaturate",                                                     ["RetVectorType"] = "Vector64",  ["RetBaseType"] = "Int32",   ["Op1VectorType"] = "Vector64",  ["Op1BaseType"] = "Int32",  ["Op2VectorType"] = "Vector64",  ["Op2BaseType"] = "Int32",                                                               ["LargestVectorSize"] = "8",  ["NextValueOp1"] = "TestLibrary.Generator.GetInt32()",         ["NextValueOp2"] = "TestLibrary.Generator.GetInt32()",                                                                                        ["ValidateIterResult"] = "Helpers.SubtractSaturate(left[i], right[i]) != result[i]"}),
```
Comment (kunalspathak) [inline src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs:1887]: why is this deleted?

File: src/coreclr/jit/hwintrinsicarm64.cpp [hunk 2]
```diff
@@ -1909,22 +1909,63 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
...
+                    }
+                }
+
+                retNode = gtNewSimdHWIntrinsicNode(retType, op1, op2, intrinsic, simdBaseJitType, simdSize);
             }
+            else
+            {
+                if (op2->TypeGet() == TYP_SIMD16)
+                {
+                    // Update the simdSize explicitly as Vector128 variant of Store() is present in AdvSimd instead of
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsicarm64.cpp:1953]: I don't see we do it currently or did I miss anything?

**Review by kunalspathak — CHANGES_REQUESTED** [review]
Added few comments.

**Comment by kunalspathak** [comment]
@SwapnilGaikwad - it seems there are some test failures:

```03:58:16.044 Running test: _AdvSimd_r::JIT.HardwareIntrinsics.Arm._AdvSimd.Program.Storex2SByte()

Assert failure(PID 85467 [0x00014ddb], Thread: 2420517 [0x24ef25]): Assertion failed '!HWIntrinsicInfo::IsInvalidNodeId(intrinsicId)' in 'JIT.HardwareIntrinsics.Arm._AdvSimd.Storex2SByteTest:RunBasicScenario_UnsafeRead():this' during 'Importation' (IL size 112; hash 0x4100adc6; MinOpts)

    File: /Users/runner/work/1/s/src/coreclr/jit/gentree.cpp:28149
    Image: /private/tmp/helix/working/A1D908D4/p/corerun

```

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs (hunk 1 above)
Comment (SwapnilGaikwad) [reply src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs:1887]: Seems accidental, I'll restore it.

File: src/coreclr/jit/hwintrinsicarm64.cpp (hunk 2 above)
Comment (SwapnilGaikwad) [reply src/coreclr/jit/hwintrinsicarm64.cpp:1953]: well, both Vector64 and Vector128 have their single vector `Store` methods in AdvSimd class. All the other Vector128 stores are in AdvSimd.Arm64 class. Thus, we cannot retrieve the simdSize from the hwintrinsiclistarm64.h for single vector stores.
Previously, we used to retrieve the simd type from the second argument that disambiguated both Vector128 and Vector64 single vector stores. Now, we cannot use the second argument as there could be a tuple. Thus, we need another mechanism to disambiguate them.

**Comment by fanyang-mono** [comment]
Test `AdvSimd_r.dll` passed locally

```
Supported ISAs:
  AdvSimd:   True
  Aes:       True
  ArmBase:   True
  Crc32:     True
  Dp:        False
  Rdm:       False
  Sha1:      True
  Sha256:    True
  Sve:       False

...

Expected: 100
Actual: 100
END EXECUTION - PASSED
```

**Review by fanyang-mono — APPROVED** [review]
Mono change is good and complete.

File: src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs (hunk 1 above)
Comment (SwapnilGaikwad) [reply src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs]: Done

**Review by kunalspathak — APPROVED** [review]
coreclr LGTM.

**Comment by kunalspathak** [comment]
/ba-g the only failure on windows/x64 is unrelated and seems flaky
